nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -213,33 +261,71 @@ class TfOpIdxMap {
 
   const std::vector<int> &getOutputs() const { return outputs; }
 
+  const std::vector<const Tensor *> &getTensors() const { return tensors; }
+
+  std::ptrdiff_t getTensorIndex(const Tensor *tensor) const {
+    auto tensor_it = std::find(tensors.begin(), tensors.end(), tensor);
+    NNTR_THROW_IF(tensor_it == tensors.cend(), std::invalid_argument)
+      << FUNC_TAG << "Cannot find index for tensor: " << tensor->getName();
+    return std::distance(tensors.begin(), tensor_it);
+  }
+
 private:
   float empty_buffer[0]; /**< reserved unintialized tensor points to this
                             buffer */
 
   std::tuple<BidirectionalIndexMap<const float *, Buffer>,   /**< buffer map
                                                               */
-             BidirectionalIndexMap<tflite::BuiltinOperator>, /**< opcode map
+             BidirectionalIndexMap<tflite::BuiltinOperator>> /**< opcode map
                                                               */
-             BidirectionalIndexMap<const float *, const Tensor *>> /**< tensor
-                                                                      map */
     maps;
 
   std::vector<int> inputs;
   std::vector<int> outputs;
+  /// since it is used as a tensor index, the order is important
+  std::vector<const Tensor *> tensors;
 };
 
-TfOpNodes buildOpNodes(const GraphRepresentation &representation) {
+TfOpNodes buildOpNodes(const GraphRepresentation &representation,
+                       flatbuffers::FlatBufferBuilder &fbb) {
   TfOpNodes nodes;
+  /// @todo TfOpNode needs to have LayerNode pointer
+  std::map<TfOpNode *, const LayerNode *> tf_to_layer;
+  std::map<const LayerNode *, TfOpNode *> layer_to_tf;
+
   /// @todo, look ahead of layers to get nodes that can be fused
   /// we will need to have a dedicated builder
   for (auto iter = representation.cbegin(); iter != representation.cend();
        iter++) {
     const auto &ln = *iter;
-    Exporter e;
+    Exporter e(&fbb);
     ln->exportTo(e, ExportMethods::METHOD_TFLITE);
 
     nodes.emplace_back(e.getResult<ExportMethods::METHOD_TFLITE>());
+    tf_to_layer.insert({nodes.back().get(), ln.get()});
+    layer_to_tf.insert({ln.get(), nodes.back().get()});
+  }
+
+  /// set arity of TfOpNodes
+  for (auto &n : nodes) {
+    auto tf_node = n.get();
+    auto layer_node = tf_to_layer.find(tf_node)->second;
+    auto layer_node_inputs = layer_node->getInputConnections();
+
+    /// assume that the TfOpNode and the LayerNode have a one-to-one
+    /// relationship
+    tf_node->arity(layer_node_inputs.size());
+    for (size_t index = 0; index < layer_node_inputs.size(); index++) {
+      auto input_layer_name = layer_node_inputs[index];
+      auto input_layer_node =
+        std::find_if(
+          representation.begin(), representation.end(),
+          [&input_layer_name](std::shared_ptr<nntrainer::LayerNode> node) {
+            return node.get()->getName() == input_layer_name;
```

Comment:
It's minor, but
``` c++
return istrequal(node.get()->getName(), input_layer_name);
```
could be better because this compares after tolower.

Reply:
@jijoongmoon I saw that function. Is there any reason lowering the case?